test/component/test_component.py

```diff
@@ -129,3 +132,36 @@ def run(self, value: int):
 
     comp = MockComponent()
     assert comp.__canals_component__
+
+
+def test_default_component_to_dict():
+    MyComponent = factory.component_class("MyComponent")
+    comp = MyComponent()
+    res = _default_component_to_dict(comp)
+    assert res == {
+        "hash": id(comp),
+        "type": "MyComponent",
+        "init_parameters": {},
+    }
+
+
+def test_default_component_from_dict():
+    MyComponent = factory.component_class("MyComponent")
+    comp = _default_component_from_dict(MyComponent, {"type": "MyComponent"})
+    assert isinstance(comp, MyComponent)
+
+
+def test_default_component_from_dict_without_type():
+    with pytest.raises(ComponentDeserializationError, match="Missing 'type' in component serialization data"):
+        _default_component_from_dict(Mock, {})
+
+
+def test_default_component_from_dict_unregistered_component(request):
+    # We use the test function name as component name to make sure it's not registered.
+    # Since the registry is global we risk to have a component with the same name registered in another test.
+    component_name = request.node.name
+
+    with pytest.raises(
+        ComponentDeserializationError, match=f"Component '{component_name}' can't be deserialized as 'Mock'"
+    ):
+        _default_component_from_dict(Mock, {"type": component_name})
```

Comment:
If I understood correctly this test checks against unregistered components only. Should we add a test for registered, but wrong components? Specifically the behavior of this call:

```python
_default_component_from_dict(ComponentA, {"type": "ComponentB"})
```

assuming both components are registered.

Reply:
It doesn't make much sense to test the component is registered. 
If I'm calling `ComponentA.from_dict(data)` for sure we imported `ComponentA` and is in the register.
If `ComponentB` is not registered we shouldn't care either as the received dictionary must be wrong.